backend/src/main/java/de/neuefische/backend/security/service/SignUpService.java

```diff
@@ -0,0 +1,32 @@
+package de.neuefische.backend.security.service;
+
+import de.neuefische.backend.security.model.CredentialsDto;
+import de.neuefische.backend.security.model.UserCredentials;
+import de.neuefische.backend.security.repo.AppUserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SignUpService {
+    AppUserRepository appUserRepo;
+
+    @Autowired
+    public SignUpService(AppUserRepository appUserRepo) {
+        this.appUserRepo = appUserRepo;
+    }
+
+    public boolean checkUsername(String username) {
+        return appUserRepo.existsById(username);
+    }
+
+    public boolean signUp(CredentialsDto credentials){
+        if(checkUsername((credentials.getUsername()))) {
+            return false;
+        }
+        BCryptPasswordEncoder bCryptPasswordEncoder = new BCryptPasswordEncoder();
```

Comment:
Use the `PasswordEncoder` bean you configured in your security context. That way there is only one point where you have to change the concrete encoder used in your app.